include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -59,9 +59,6 @@ struct aws_s3_request {
     /* Owning meta request. */
     struct aws_s3_meta_request *meta_request;
 
-    /* Current retry token for the request. If it has never been retried, this will be NULL. */
```

Comment:
did this behavior change? you need a token for every request, or you'll never recover from an outage.

Reply:
meaning, you get a token even if it's the first request, and then use it for retries if you need to. otherwise, release it when you're finished.

Reply:
The retry token will still be one per request, but now it's attached to the owning aws_s3_vip_connection that the request is being done on.  This is so that we can back off the actual connection that's doing the request, and also have an easier way to do a retry for an error when just trying to make the connection.

Reply:
Re-worked to always try grabbing a token when a request begins.